src/commands/optimize.ts

```diff
@@ -0,0 +1,163 @@
+import { Flags } from '@oclif/core';
+import { Optimizer, Output, Report, ReportElement } from '@asyncapi/optimizer';
+import Command from '../base';
+import { ValidationError } from '../errors/validation-error';
+import { load } from '../models/SpecificationFile';
+import * as inquirer from 'inquirer';
+import chalk from 'chalk';
+import { promises } from 'fs';
+import { Example } from '@oclif/core/lib/interfaces';
+const { writeFile } = promises;
+
+export enum Optimizations {
+  REMOVE_COMPONENTS='remove-components',
+  REUSE_COMPONENTS='reuse-components',
+  MOVE_TO_COMPONETS='move-to-components'
+}
+
+export enum Outputs {
+  TERMINAL='terminal',
+  NEW_FILE='new-file',
+  OVERWRITE='overwrite'
+}
+export default class Optimize extends Command {
+  static description = 'optimize asyncapi specification file';
+  isInteractive = false;
+  optimizations?: Optimizations[];
+  outputMethod?: Outputs;
+
+  static examples: Example[] = [
+    'asyncapi optimize ./asyncapi.yaml',
+    'asyncapi optimize ./asyncapi.yaml --no-tty',
+    'asyncapi optimize ./asyncapi.yaml --optimization=remove-components,reuse-components,move-to-components --no-tty',
+    'asyncapi optimize ./asyncapi.yaml --optimization=remove-components,reuse-components,move-to-components --output=terminal --no-tty',
+  ];
+  
+  static flags = {
+    help: Flags.help({ char: 'h' }),
+    optimization: Flags.string({char: 'p', default: Object.values(Optimizations), options: Object.values(Optimizations), multiple: true, description: 'select the type of optimizations that you want to apply.'}),
+    output: Flags.string({char: 'o', default: Outputs.TERMINAL, options: Object.values(Outputs), description: 'select where you want the output.'}),
+    'no-tty': Flags.boolean({ description: 'do not use an interactive terminal', default: false }),
+  };
+
+  static args = [
+    { name: 'spec-file', description: 'spec path, url, or context-name', required: false },
+  ];
+
+  async run() {
+    const { args, flags } = await this.parse(Optimize); //NOSONAR
+    const filePath = args['spec-file'];
+    const specFile = await load(filePath);
+    this.isInteractive = !flags['no-tty'];
+    this.optimizations = flags.optimization as Optimizations[];
+    this.outputMethod = flags.output as Outputs;
+    const optimizer = new Optimizer(specFile.text());
+    const report = await optimizer.getReport();
+    const isURL = !!specFile.getFileURL();
+
+    if (!(report.moveToComponents?.length || report.removeComponents?.length || report.reuseComponents?.length)) {
+      if (isURL) {
+        this.log(`URL ${specFile.getFileURL()} looks optimized!`);
+      } else {
+        this.log(`File ${specFile.getFilePath()} looks optimized!`);
```

Comment:
also resulting file should be specified, its location, so from console link you can quickly access it, especially when calling `asyncapi optimize ./asyncapi.yaml --no-tty`

Reply:
this message is shown when the file can't be optimized since there are no optimizations to be applied. ( I will make the message a little clearer.)